admin-dev/themes/new-theme/js/pages/product/edit/product-type-switcher.ts

```diff
@@ -57,7 +57,12 @@ export default class ProductTypeSwitcher {
 
     this.productId = parseInt($productForm.data('productId'), 10);
     this.initialType = <string> this.$typeSelector.val();
+    this.registerProductTypePreviewOnClickEvent();
+  }
 
+  public registerProductTypePreviewOnClickEvent(): void {
```

Comment:
I don't think this should be part of the public API, especially here you disabled/reenable the interactions of two components inside the updater but the updater shouldn't need to know how to handle those interactions

The component itself is responsible for it and could have a public API enable/disable (which is used by the updater), that handles disabling or blocking the click events.

Seems like the preview and the type switcher need to be aware of each other, either via a direct dependency (one is injected into another) or via some events, but the injection is fine I think (since the switcher is also used in another context like creation modal it should remain independent, but it can be injected as a dependency of the preview component however).